javatests/google/registry/export/ExportConstantsTest.java

```diff
@@ -56,38 +58,79 @@
   public void testBackupKinds_matchGoldenBackupKindsFile() throws Exception {
     URL goldenBackupKindsResource =
         getResource(ExportConstantsTest.class, GOLDEN_BACKUP_KINDS_FILENAME);
-    final Pattern stripComments = Pattern.compile("\\s*#.*$");
-    List<String> goldenKinds = FluentIterable
-        .from(Splitter.on('\n').split(
-            Resources.toString(goldenBackupKindsResource, UTF_8).trim()))
-        .transform(
-            new Function<String, String>() {
-              @Override @Nullable public String apply(@Nullable String line) {
-                return stripComments.matcher(line).replaceFirst("");
-              }})
-        .toList();
+    List<String> goldenKinds = getGoldenKinds(Resources.toString(
+        goldenBackupKindsResource, UTF_8));
     ImmutableSet<String> actualKinds = ExportConstants.getBackupKinds();
-    String updateInstructions = String.format(
-        UPDATE_INSTRUCTIONS_TEMPLATE,
-        goldenBackupKindsResource.toString(),
-        Joiner.on('\n').join(actualKinds));
+    String updateInstructions = getUpdateInstructions("backed-up",
+        goldenBackupKindsResource.toString(), actualKinds );
     assertWithMessage(updateInstructions)
         .that(actualKinds)
         .containsExactlyElementsIn(goldenKinds)
         .inOrder();
   }
 
   @Test
+  public void testReportingKinds_matchGoldenReportingKindsFile() throws Exception {
+    URL goldenReportingKindsResource =
+        getResource(ExportConstantsTest.class, GOLDEN_REPORTING_KINDS_FILENAME);
+    List<String> goldenReportingKinds = getGoldenKinds(Resources.toString(
+        goldenReportingKindsResource, UTF_8));
+    ImmutableSet<String> actualKinds = ExportConstants.getReportingKinds();
+    String updateInstructions = getUpdateInstructions("reporting",
+        goldenReportingKindsResource.toString(), actualKinds );
+    assertWithMessage(updateInstructions)
+        .that(actualKinds)
+        .containsExactlyElementsIn(goldenReportingKinds)
+        .inOrder();
+  }
+
+  @Test
   public void testReportingKinds_areSubsetOfBackupKinds() throws Exception {
     assertThat(ExportConstants.getBackupKinds()).containsAllIn(ExportConstants.getReportingKinds());
   }
 
-  @Test
-  public void testReportingEntityClasses_areAllBaseEntityClasses() throws Exception {
-    for (Class<? extends ImmutableObject> clazz : ExportConstants.REPORTING_ENTITY_CLASSES) {
-      assertThat(clazz.isAnnotationPresent(Entity.class))
-          .named(String.format("class %s is an @Entity", clazz.getSimpleName()))
-          .isTrue();
-    }
+  @Test 
+  public void testReportingKinds_areSubsetOfEntityClasses() throws Exception {  
+    ImmutableSet<String> entityClasses =  FluentIterable.from(EntityClasses.ALL_CLASSES) 
```

Comment:
You somehow managed to insert two "'LINE SEPARATOR' (U+2028)" Unicode characters in this function which are thoroughly messing up my IDE.  You can see the invalid characters on a text-based command-line git diff.  Please remove them.